parts/linux/cloud-init/artifacts/reconcile-private-hosts.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+set -o nounset
+set -o pipefail
+SLEEP_SECONDS=15
+clusterFQDN={{GetParameter "kubernetesEndpoint"}}
+if [[ $clusterFQDN != *.privatelink.* ]]; then
+  echo "skip reconcile hosts for $clusterFQDN since it's not AKS private cluster"
+  exit 0
+fi
+echo "clusterFQDN: $clusterFQDN"
+function get-apiserver-ip-from-tags() {
```

Comment:
can you move the function declaration to the end?

Reply:
I have to define this function first and call it within the while loop.

Reply:
then can you move it to the very beginning? It's better to separate declaration and usage of the functions.

Reply:
sure.